File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp ;
        int n = nums.size();
        vector<int> v ;
        for(int i = 0 ; i < n ; i++){
            if(mp[target - nums[i]] > 0 ){
                v.push_back(i);
                v.push_back(mp[target - nums[i]] - 1);
            }
            mp[nums[i]] = i + 1;
        }
        return v ;
    }
};

File: 0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] a , int target){
        int[] res = new int[2];
        Map<Integer,Integer> mp = new HashMap<>();
        for(int i = 0 ; i < a.length ; i++){
            if(mp.containsKey(target - a[i])){
                res[1] = i ;
                res[0] = mp.get(target - a[i]);
            }
            mp.put(a[i],i);
        }
        return res ;
    }
}
